### PR TITLE
drivers: linux: axi_io: Add busybox devmem support

### PR DESCRIPTION
**Pull request by dbogdan**
"busybox devmem" is a tool that can be used to read/write from physical address.
The patch extends the linux/axi_io.c implementation to use this tool as needed
instead of a UIO device.
This feature can be enabled by defining a DEVMEM symbol (e.g., "-DDEVMEM").

Note: "busybox" package is mandatory to be present(e.g., "apt-get install busybox").

Signed-off-by: Dragos Bogdan <dragos.bogdan@analog.com>

### CODE REVIEW

**Comment by dbogdan** [comment]
v2:
 - Fix astyle reported issues

File: drivers/platform/linux/axi_io.c [hunk 1]
```diff
@@ -111,15 +112,65 @@ static int32_t axi_io_read_write(uint32_t base, uint32_t offset, uint32_t *read,
...
+	if (write)
+		ret = snprintf(command, sizeof(command),
+			       "busybox devmem 0x%x w 0x%x", base + offset, *write);
+	else if (read)
+		ret = snprintf(command, sizeof(command),
+			       "busybox devmem 0x%x", base + offset);
+	else
+		return FAILURE;
+
+	if (ret < 0 && ret >= (int32_t) sizeof(command))
```
Comment (buha) [inline drivers/platform/linux/axi_io.c:139]: shouldn't this be || ?

Comment (dbogdan) [reply drivers/platform/linux/axi_io.c:139]: Yes, it should.

**Review by buha — APPROVED** [review]

**Comment by dbogdan** [comment]
v3:
 - Logical operator fix.

**Pull request merged**